src/postgres_mcp/server.py

```diff
@@ -16,92 +18,168 @@
 # Constants
 PG_STAT_STATEMENTS = "pg_stat_statements"
 HYPOPG_EXTENSION = "hypopg"
+MAX_RETRIES = 3
+RETRY_DELAY = 1  # seconds
 
 ResponseType = List[types.TextContent | types.ImageContent | types.EmbeddedResource]
 
+logger = logging.getLogger(__name__)
+
+
+def obfuscate_password(url: str) -> str:
```

Comment:
We should add a handful of tests for this.

Reply:
Added tests.